include/zephyr/instrumentation/instrumentation_ringbuffer.h

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2023 Linaro
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_INSTRUMENTATION_RINGBUFFER_H_
+#define ZEPHYR_INCLUDE_INSTRUMENTATION_RINGBUFFER_H_
+
+#include <stdio.h>
+#include <inttypes.h>
+#include <assert.h>
+#include <string.h>
+
+#include <zephyr/instrumentation/instrumentation.h>
+
+#define DEBUG 0
+
+void init_instr_ringbuffer(void);
+
+struct instr_record* put_item_claim(enum instr_event_types type);
```

Comment:
these symbols should likely be prefixed with the subsys name

Reply:
Done. I've prefixed them with `instr_rb_`